src/canvaslms/cli/calendar.nw

```diff
@@ -0,0 +1,252 @@
+@
+
+
+\section{Getting and outputting calendar events}
+
+We fetch the calendar events and output them in CSV format.
+<<get and output calendar events>>=
+try:
+  # Prepare parameters for API call
+  params = {}
+  if context_codes:
+    params['context_codes'] = context_codes
+  if args.start_date:
+    params['start_date'] = args.start_date
+  if args.end_date:
+    params['end_date'] = args.end_date
+  if args.type:
+    params['type'] = args.type
+    
+  events = canvas.get_calendar_events(**params)
+  
+  for event in events:
+    <<output calendar event data>>
+    
+except canvasapi.exceptions.CanvasException as e:
+  canvaslms.cli.err(1, f"Failed to get calendar events: {e}")
+@
+
+We output the calendar event data in a structured format.
+<<output calendar event data>>=
+# Get context information
+context_type = getattr(event, 'context_type', '')
+context_name = getattr(event, 'context_name', '')
+
+# Format dates
+start_at = getattr(event, 'start_at', '')
+end_at = getattr(event, 'end_at', '')
+
+row = [
+  getattr(event, 'id', ''),
+  getattr(event, 'title', ''),
+  start_at,
+  end_at,
+  context_type,
+  context_name
+]
+output.writerow(row)
+@
+
+
+\section{The \texttt{calendar show} subcommand}
+
+The \texttt{calendar show} subcommand shows details of a specific calendar event.
+<<functions>>=
+def add_calendar_show_command(subp):
+  """Adds the calendar show subcommand and its options to argparse subparser subp"""
+  calendar_show_parser = subp.add_parser("show",
+    help="Shows details of a specific calendar event",
+    description="Shows details of a specific calendar event")
+  calendar_show_parser.set_defaults(func=calendar_show_command)
+  <<add calendar show arguments>>
+
+def calendar_show_command(config, canvas, args):
+  """Shows details of a specific calendar event"""
+  <<get and display calendar event details>>
+@
+
+The show command requires an event ID.
+<<add calendar show arguments>>=
+calendar_show_parser.add_argument("event_id",
+  help="The ID of the calendar event to show")
+@
+
+We fetch and display the event details.
+<<get and display calendar event details>>=
+try:
+  event = canvas.get_calendar_event(args.event_id)
+  
+  print(f"Event ID: {event.id}")
+  print(f"Title: {getattr(event, 'title', 'N/A')}")
+  print(f"Description: {getattr(event, 'description', 'N/A')}")
+  print(f"Start: {getattr(event, 'start_at', 'N/A')}")
+  print(f"End: {getattr(event, 'end_at', 'N/A')}")
+  print(f"Location: {getattr(event, 'location_name', 'N/A')}")
+  print(f"Context: {getattr(event, 'context_type', 'N/A')} - {getattr(event, 'context_name', 'N/A')}")
+  print(f"URL: {getattr(event, 'html_url', 'N/A')}")
+  
+except canvasapi.exceptions.CanvasException as e:
+  canvaslms.cli.err(1, f"Failed to get calendar event: {e}")
+@
+
+
+\section{The \texttt{calendar create} subcommand}
+
+The \texttt{calendar create} subcommand creates a new calendar event.
```

Comment:
There are multiple types of events in the Canvas calendar. We want to be able to create bookable time slots events too. Please add that.